src/WingetCreateCLI/Commands/UpdateCommand.cs

```diff
@@ -144,6 +156,26 @@ public override async Task<bool> Execute()
                     this.Id = exactId;
                 }
 
+                if (!string.IsNullOrEmpty(this.ReplaceVersion))
+                {
+                    // If update version is same as replace version, it's a regular update.
+                    if (this.Version == this.ReplaceVersion)
+                    {
+                        this.Replace = false;
+                    }

```

Comment:
I think it would be better to show an error message to the user that the replacement version matches the update version since that should not be a supported scenario. That way if the user accidentally types in an identical replacement version, they are notified rather than performing an update.